sdk/app_cpu1/common/drv/eddy_current_sensor.c

```diff
@@ -9,36 +9,103 @@ void eddy_current_sensor_init(void)
+    uint32_t config_reg_address = base_addr + (3 * sizeof(uint32_t));
+    Xil_Out32(config_reg_address, (Xil_In32(config_reg_address) | 0x3));
+}
 
-    eddy_current_sensor_set_divider(base_addr, divider - 1);
+void eddy_current_sensor_trigger_on_pwm_clear(uint32_t base_addr)
+{
+    // Get the current value of the config register and clear both the pwm_high and pwm_low trigger bits
+    uint32_t config_reg_address = base_addr + (3 * sizeof(uint32_t));
+    Xil_Out32(config_reg_address, (Xil_In32(config_reg_address) & ~0x3));
 }
 
-void eddy_current_sensor_set_divider(uint32_t base_addr, uint8_t divider)
+void eddy_current_sensor_set_timing(uint32_t base_addr, uint32_t sclk_freq_khz, uint32_t propogation_delay_ns)
 {
-    Xil_Out32(base_addr + (2 * sizeof(uint32_t)), divider);
+    // 10 MHz is max frequency (faster frequencies limited by diff/single transceivers)
+    //   should give sclk_cnt = 10 axi cycles
+    if (sclk_freq_khz > 10000) {
+        sclk_freq_khz = 10000;
+    }
+
+    // 500 kHz is min frequency (slower frequencies won't complete in a PWM carrier cycle)
+    //   should give sclk_cnt = 200 axi cycles
+    if (sclk_freq_khz < 500) {
+        sclk_freq_khz = 500;
+    }
+
+    // This is period in ns for one half of the sclk period
+    // We want half a period since sclk_cnt is the number of AXI CLK cycles to wait before toggling SCLK
+    uint32_t sclk_half_period_ns = (1000000 / sclk_freq_khz) / 2;
+
+    uint32_t axi_period_ns = 1000 / AXI_CLK_FREQ_MHZ;
+
+    uint32_t sclk_cnt = sclk_half_period_ns / axi_period_ns;
+
+    Xil_Out32(base_addr + (2 * sizeof(uint32_t)), sclk_cnt);
+
+    // SHIFT delayer
+    //   Why is this needed? The Kaman adapter board's filtering introduces a significant
+    //   propogation delay into the system. On the FPGA side, the SCLK signal
+    //   being generated will fall, which is when we would like to sample the MISO line.
+    //   However, the fall of SCLK will take a while to propogate through the adapter
+    //   board (270ns for example) and then the valid data on the MISO lines will take a
+    //   while (again, 270ns for example) to propogate back. In the example, this is
+    //   a total round-trip propogation delay of 540ns. The actual delay depends on the
+    //   RC filters used on the Kaman adapter board.
+    //
+    //   This code takes the ONE-WAY propogation delay in nanoseconds of the Kaman adapter
+    //   board's filtering, doubles it for the round-trip propogation delay, and then
+    //   adds half of the user-requested SCLK period so that the shifting occurs halfway
+    //   through when the bit is valid. See the amdc_spi_master.v for details on how
+    //   the shift signal is propogated through a shift register and then the appropriate delay
+    //   is selected by the shift_index value calculated below.
+    uint32_t delay_time = (2 * propogation_delay_ns) + (1 * sclk_half_period_ns);
+
+    uint32_t shift_index = delay_time / axi_period_ns;
```

Comment:
Let's not refer to the FPGA clock freq as the AXI clock frequency, since in general, they could be different. From the C code, let's refer to the digital logic clock freq as the "FPGA clock frequency" and remove the mentions of "AXI"